### PR TITLE
FIxed is_on_constant_path() using in all places

### PR DESCRIPTION
**Pull request by antonvor**
Issues: 117928, 117058
FIx for https://github.com/openvinotoolkit/openvino/pull/19141

PR to release branch: https://github.com/openvinotoolkit/openvino/pull/19256

### CODE REVIEW

**Comment by antonvor** [comment]
@itikhono @dmitry-gorokhov take a look please?

File: src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp [hunk 1]
```diff
@@ -254,7 +254,7 @@ bool isSuitableChildForFusingMatMul(const std::shared_ptr<const Node> &node, con
     ov::PartialShape matmul_shape;
     for (const auto &parent_out : node->input_values()) {
         const auto parent = parent_out.get_node_shared_ptr();
-        if (ov::op::util::is_on_constant_path(parent)) {
+        if (ov::op::util::is_on_constant_path(parent->output(0))) {
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp:257]: Shouldn't it be `parent_out` instead?
As far as I understand there is no guarantee `parent` is connected with `node` via 0th port.

**Comment by v-Golubev** [comment]
@antonvor [this line](https://github.com/openvinotoolkit/openvino/blob/95300ac5bfc3f60ded81aa3425b550421d88efbe/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp#L268) should be also fixed in the same way: we should call `is_on_constant_path` for output, not node

File: src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp (hunk 1 above)
Comment (antonvor) [reply src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp]: changed to `parent_out`, thanks

**Comment by antonvor** [comment]
@v-Golubev thank you for your comment, fixed
https://github.com/openvinotoolkit/openvino/pull/19239/files#diff-c19b63cc114defd4e10f8dc600407fc4c3d6a5e38d111116a732bc5017a28623R268

**Comment by itikhono** [comment]
Could you fix other "is_on_constant_path" usages? maybe in the next PR 
image

File: src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp [hunk 2]
```diff
@@ -254,7 +254,7 @@ bool isSuitableChildForFusingMatMul(const std::shared_ptr<const Node> &node, con
     ov::PartialShape matmul_shape;
     for (const auto &parent_out : node->input_values()) {
         const auto parent = parent_out.get_node_shared_ptr();
-        if (ov::op::util::is_on_constant_path(parent)) {
```
Comment (itikhono) [inline src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp:257]: should we add some test for this case?

**Review by itikhono — APPROVED** [review]

**Comment by antonvor** [comment]
> Could you fix other "is_on_constant_path" usages? maybe in the next PR image

I fixed check in `ConvertMatMulToFC()` pass only, let's update other places in separate PR

File: src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp (hunk 2 above)
Comment (antonvor) [reply src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp:257]: added `SplitMatMulConcat` subgraph test for this case

**Review by v-Golubev — APPROVED** [review]

**Pull request merged**